ds3-sdk/src/main/java/com/spectralogic/ds3client/commands/interfaces/AbstractRequest.java

```diff
@@ -27,13 +27,13 @@
 
 public abstract class AbstractRequest implements Ds3Request {
 
-    private final Multimap<String, String> headers = buildDefaultHeaders();
+    private final RequestHeaders headers = buildDefaultHeaders();
     private final Map<String, String> queryParams = new HashMap<>();
     
-    private static Multimap<String, String> buildDefaultHeaders() {
+    private static RequestHeaders buildDefaultHeaders() {
         final Multimap<String, String> headers = TreeMultimap.create();
```

Comment:
Now that we have the RequestHeaders class, it should be responsible for creating the storage that it needs, rather than the consumer of it.  Move the TreeMultimap creation into the RequestHeaders class, and then use the new header insertion method.